rfcs/text/0049/entity.yml

```diff
@@ -0,0 +1,115 @@
+---
+# Entity field set schema definition
+name: entity
+title: Entity
+group: 2
+type: group
+short: Fields to describe various types of entities across IT environments.
+description: >
+  The entity fields provide a standardized way to represent and categorize
+  different types of components within an IT environment, including those
+  that don't have dedicated field sets in ECS. An entity represents a discrete,
+  identifiable component that can be described by a set of attributes and
+  maintains its identity over time.
+root: true
+fields:
+
+  - name: entity.id
+    level: core
+    type: keyword
+    short: Unique identifier for the entity.
+    description: >
+      A unique identifier for the entity. When multiple identifiers exist, this should be
+      the most stable and commonly used identifier that: 1) persists across the entity's
+      lifecycle, 2) ensures uniqueness within its scope, 3) is commonly used for queries
+      and correlation, and 4) is readily available in most observations (logs/events).
+      For entities with dedicated field sets (e.g., host, user), this value should match
+      the corresponding *.id field. Alternative identifiers (e.g., ARNs values in AWS, URLs)
+      can be preserved in entity.raw.
+    example: i-04ff5d36be3d6896c, arn:aws:s3:::my-bucket, projects/123456789/locations/us-central1/instances/my-db
+
+  - name: entity.source
+    level: core
+    type: keyword
+    short: Source module or integration that provided the entity data.
+    description: >
```

Comment:
I think the document source can usually be determined from `event.dataset` or the index its in. Do you know of any cases where the existing data isn't enough to determine the source?

Reply:
For entity-centric documents whose metadata we captured from multiple source documents, the idea is that this entity.source field will capture multiple values (a multivalue keyword field). Related to my comment here https://github.com/elastic/ecs/pull/2461#discussion_r2051854253

I propose we retain this for Stage 1 and revisit the decision in subsequent RFCs after further testing and requirements gathering.